### PR TITLE
dataflow tests: simplify wait function + fix landsat prime flaky test

### PR DESCRIPTION
**Pull request by davidcavazos**
## Description

No change in functionality, but only make one of the utility functions simpler and easier to maintain.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Review by parthea — APPROVED** [review]

File: dataflow/gpu-examples/tensorflow-landsat-prime/e2e_test.py [hunk 1]
```diff
@@ -60,8 +60,8 @@ def test_tensorflow_landsat(
     utils: Utils, bucket_name: str, run_dataflow_job: str
 ) -> None:
     # Wait until the job finishes.
-    timeout = 30 * 60  # 30 minutes
-    utils.dataflow_jobs_wait(job_name=utils.hyphen_name(NAME), timeout_sec=timeout)
+    job_id = utils.dataflow_job_id(utils.hyphen_name(NAME))
+    utils.dataflow_jobs_wait(job_id, timeout_sec=30 * 60)
```
Comment (nicain) [inline dataflow/gpu-examples/tensorflow-landsat-prime/e2e_test.py:64]: Consider changing 30 * 60 to a variable to avoid a magic number.

**Review by nicain — APPROVED** [review]
LGTM

File: dataflow/gpu-examples/tensorflow-landsat-prime/e2e_test.py (hunk 1 above)
Comment (davidcavazos) [reply dataflow/gpu-examples/tensorflow-landsat-prime/e2e_test.py]: Thanks, made into a constant at the top.

**Comment by davidcavazos** [comment]
Transient error, quota exceeded for CPUs, rerunning tests.

```
QUOTA_EXCEEDED: Instance 'df-hvm-pool-0-2022-02-15-15-00-07-326717310653860612-w7b9' creation failed: Quota 'CPUS' exceeded. Limit: 200.0 in region us-central1.
```

**Comment by davidcavazos** [comment]
Looks like another timeout but the test passed.

```
e2e_test.py::test_tensorflow_landsat PASSED                              [100%]

- generated xml file: /workspace/dataflow/gpu-examples/tensorflow-landsat-prime/sponge_log.xml -
======================== 1 passed in 1862.14s (0:31:02) ========================
nox > Session py-3.8 was successful.


err: signal: killed
```

**Comment by davidcavazos** [comment]
Looks like there CPU worker pools and the GPU worker pools were passing data around through different functions, causing a lot of temporary file reading/writing since the image files are quite large. Manually fusing all the functions for the GPU workers into a single function so they don't have to write temporary files and everything happens in memory.

**Pull request merged**